packages/daemon/src/identity/PeerUserIdentity.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+import { mainframeIDType, type MainframeID } from '@mainframe/data-types'
+
+import Identity from './Identity'
+
+type FeedTypes = 'public' // TODO: define more types
+type FeedHash = string
+
+export type PeerData = {
+  name: string,
+  avatar?: ?string,
+}
+
+export type Feeds = { [type: FeedTypes]: FeedHash }
+
+export type PeerUserIdentitySerialized = {
+  id: string,
+  feeds: Feeds,
+  data: PeerData,
+}
+
+export default class PeerUserIdentity extends Identity {
+  static fromJSON = (params: PeerUserIdentitySerialized): PeerUserIdentity =>
+    new PeerUserIdentity(mainframeIDType(params.id), params.data, params.feeds)
+
+  static toJSON = (peer: PeerUserIdentity): PeerUserIdentitySerialized => ({
+    id: peer._id,
+    feeds: peer._feeds,
+    data: {
+      name: peer._name,
+      avatar: peer._avatar,
+    },
+  })
+
+  _feeds: Feeds
+  _name: string
```

Comment:
I think we should assume the name might not be provided, so type it as `?string`

Reply:
sounds good, will make this change in the next PR